src/migrate/index.js

```diff
@@ -108,9 +115,49 @@ export default class Migrator {
     });
   }
 
+  _createMigrationLockTable(tableName) {
+    return this.knex.schema.createTable(tableName, function(t) {
+      t.boolean('is_locked');
+    });
+  }
+
+  _getLockTableName(tableName) {
+    return tableName + '_lock';
+  }
+
+  _isLocked() {
+    var tableName = this._getLockTableName(this.config.tableName);
+    return this.knex(tableName)
+      .count('is_locked as count')
+      .then(function(data) {
+        if (data[0].count > 0) {
+          return true;
+        }
+        return false;
+      });
+  }
+
+  _lockMigrations() {
```

Comment:
This probably still allows multiple connections to take lock in the same time and it just creates multiple rows... 

You could initialize table with single row and use transaction to make sure you were the only one who read and set the value of that row e.g. with http://knexjs.org/#Builder-forUpdate